docs/explanations/benchmarking.md

```diff
@@ -1,9 +1,50 @@
 # Benchmarking
 
-Periodically we will be publishing results of benchmarking the Synapse Python Client compared to directly working with AWS S3. The purpose of these benchmarks is to make data driven decisions on where to spend time optimizing the client. Additionally, it will give us a way to measure the impact of changes to the client.
+Periodically we will be publishing results of benchmarking the Synapse Python Client
+compared to directly working with AWS S3. The purpose of these benchmarks is to make
+data driven decisions on where to spend time optimizing the client. Additionally, it
+will give us a way to measure the impact of changes to the client.
 
 ## Results
 
+### 04/01/2024: Uploading files to Synapse
+These benchmarking results bring together some important updates to the Upload logic. It
+has been re-written to bring a focus to concurrent file uploads and more effecient use
+of available threads. As a result of this change it is not reccommended to
+increase `max_threads` manually. Based on the available CPU cores this python package
+will use `multiprocessing.cpu_count() + 4`. For this testing the default thread size
+for the machine testing took place on was `6`.
+
+The results were created on a `t3a.micro` EC2 instance with a 200GB disk size running
+in us-east-1. The script that was run can be found in `docs/scripts/uploadBenchmark.py`.
+
+
+#### Some insights:
+- The use of the
+[Object-Orientated interfaces/Models Interface](../reference/oop/models.md) results in
+the best performance out of the box and will scale based on the hardware it's run on.
+- Increasing the number of threads did increase performance for the previous upload
```

Comment:
I felt like the full piece gave a decent disclaimer:
```
- Increasing the number of threads did increase performance for the previous upload
logic and Synapseutils functionality. It may also increase performance for new uploads,
however, it was not tested again. Increasing the number of threads in use also has
inconsistent stability as found in previous benchmarks.
```

In particular:
```
Increasing the number of threads in use also has inconsistent stability as found in previous benchmarks.
```

This is a "change at your own risk" kind of thing.